lib/matplotlib/axes/_axes.py

```diff
@@ -5820,9 +5820,13 @@ def _pcolorargs(self, funcname, *args, shading='auto', **kwargs):
                                 f" see help({funcname})")
         else:    # ['nearest', 'gouraud']:
             if (Nx, Ny) != (ncols, nrows):
-                raise TypeError('Dimensions of C %s are incompatible with'
-                                ' X (%d) and/or Y (%d); see help(%s)' % (
-                                    C.shape, Nx, Ny, funcname))
+                if (Nx, Ny) == (ncols + 1, nrows + 1):
+                    X = (X[1:, 1:] + X[:1, :-1]) / 2
+                    Y = (Y[1:, :-1] + Y[:-1, :-1]) / 2
```

Comment:
I haven't looked at these shapes in detail, but shouldn't these be consistent?
```suggestion
                    X = (X[1:, 1:] + X[:-1, :-1]) / 2
                    Y = (Y[1:, 1:] + Y[:-1, :-1]) / 2
```

Reply:
Might work, but I thought is made more since to change the X and Y boundary points independently.